### PR TITLE
Add HDInsight Autoscale schedule based

### PR DESCRIPTION
**Pull request by AliciaLiMicrosoft**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*Add Schedule based Autoscale for HDInsight
*
*

### CODE REVIEW

File: 101-hdinsight-autoscale-schedulebased/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# HDInsight (Spark + Schedule based Autoscale)
+
+<IMG SRC="https://azurequickstartsservice.blob.core.windows.net/badges/101-hdinsight-custom-ambari-db/PublicLastTestDate.svg" />&nbsp;
```
Comment (bmoore-msft) [inline 101-hdinsight-autoscale-schedulebased/README.md:3]: see pipeline results for proper button format

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  "parameters": {
+    "clusterName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the HDInsight cluster to create."
+      }
+    },
+    "clusterLoginUserName": {
+      "type": "string",
+      "defaultValue": "admin",
```
Comment (bmoore-msft) [inline 101-hdinsight-autoscale-schedulebased/azuredeploy.json:13]: remove defaultValue

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,228 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterName": {
```
Comment (bmoore-msft) [inline 101-hdinsight-autoscale-schedulebased/azuredeploy.json:5]: add a defaultValue using uniqueString()

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    },
+    "clusterLoginPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+      }
+    },
+    "sshUserName": {
+      "type": "string",
+      "defaultValue": "sshuser",
```
Comment (bmoore-msft) [inline 101-hdinsight-autoscale-schedulebased/azuredeploy.json:26]: remove defaultValue

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+      "name": "[uniqueString(resourceGroup().id)]",
+      "type": "Standard_LRS"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('defaultStorageAccount').name]",
+      "location": "[parameters('location')]",
+      "apiVersion": "2016-01-01",
```
Comment (bmoore-msft) [inline 101-hdinsight-autoscale-schedulebased/azuredeploy.json:56]: check pipeline results for BP violations

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+            }
+          }
+        },
+        "storageProfile": {
+          "storageaccounts": [
+            {
+              "name": "[replace(replace(reference(resourceId('Microsoft.Storage/storageAccounts', variables('defaultStorageAccount').name), '2016-01-01').primaryEndpoints.blob,'https://',''),'/','')]",
+              "isDefault": true,
+              "container": "[parameters('clusterName')]",
+              "key": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('defaultStorageAccount').name), '2016-01-01').keys[0].value]"
```
Comment (bmoore-msft) [inline 101-hdinsight-autoscale-schedulebased/azuredeploy.json:92]: update apiVersion

File: 101-hdinsight-autoscale-schedulebased/README.md (hunk 1 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-autoscale-schedulebased/README.md]: Updated README and metadata.json.

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json (hunk 2 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-autoscale-schedulebased/azuredeploy.json]: removed

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json (hunk 3 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-autoscale-schedulebased/azuredeploy.json:5]: added

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json (hunk 4 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-autoscale-schedulebased/azuredeploy.json]: removed

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json (hunk 6 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-autoscale-schedulebased/azuredeploy.json]: updated

File: 101-hdinsight-autoscale-schedulebased/azuredeploy.json (hunk 5 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-autoscale-schedulebased/azuredeploy.json]: Now the only remaining issue with best practice is  Depends On Must not start with [concat(

I have tried mutliple ways, still can't fix this issue. @bmoore-msft , can you please suggest how to fix this issue? Thanks

**Pull request merged**